buildtest/settings/slurm-example.yml

```diff
@@ -0,0 +1,14 @@
+editor: vi
+executors:
+  normal:
+    type: slurm
+    description: submit jobs with slurm using srun instead of sbatch
+    launcher: sbatch
```

Comment:
Launcher needs to be under vars as it is an executor specific variable.

Reply:
``vars`` is an array that can have any key/value pair. The ``launcher`` needs to be of enum restricted to few types.  I may misunderstood, the ``vars`` are for defining shell variables, if its for executor specific vars, how do you intend on using it.

Reply:
Okay that makes sense, but you'll also need to update the executor class [here](https://github.com/HPC-buildtest/buildtest-framework/blob/a5ab2e7d1f47c99fa936c646c4051f4280964b71/buildtest/executors/base.py#L105) that is expecting launcher under vars. I'm not sure if your recipes will validate, but if you report that they do, it definitely is easier to have the slurm type validate for launcher. Perhaps there should be a default for sbatch if it's not defined, and then don't require the user to specify it in the config?

And in this case of defining specific variables, it's not clear to me what purpose "vars" and "options" serve?

Reply:
If "vars" refers to shell variables then they would better be called environment or something like that. Vars is way too general to be clear to the user what it's referring to (I still don't get it).

Reply:
How about let's remove ``vars`` it may be confusing both of us and I dont have a clear idea how it will be used yet

Reply:
the ``options`` is a list of ``SBATCH`` directives that can get injected into the test script. And there can be way to override some of the options in the test configuration. buildtest should add ``#SBATCH`` in front of each item, so that it is used as intended, user shouldn't put any arbitrary value in here.

Reply:
Removing vars sounds good to me! Environment is fairly straight forward, but we have it represented in the [test config](https://github.com/HPC-buildtest/schemas/blob/master/script/script-v0.0.1.schema.json#L27). We'll need to think more about this setup - for example, hard coding environment variables into the testing file isn't the best idea if you want to put them on GitHub. It might also be the case that the runner would want to specify a file name instead, or have a config environment section to override the local one...

Anyway I agree we don't have a good sense of these things, so let's remove vars for now, both for the default config, and the get_variables function.  So you probably want something like:

```python
    def load(self, name):
        """Load the executor preferences from the provided config, which is
           added and indexed with "name." For slurm we look for the following
           in vars:
           :param launcher: defaults to sbatch
           :type launcher: string
        """
        self.launcher = self._settings.get("launcher", "sbatch")
```

Reply:
Let's see if Travis passes all regtest. It worked locally. I made all the changes you requested.